### PR TITLE
Document posts api

### PR DESCRIPTION
**Pull request by madisonsites**
@JoshSmith [Issue #415](https://github.com/code-corps/code-corps-api/issues/415)

I didn't create a multiple posts response just yet, but wanted a feedback on how this is thus far... 😸

### CODE REVIEW

File: blueprint/api.apib [hunk 1]
```diff
@@ -141,6 +141,84 @@ Exchange credentials for `access_token`.
...
+
++ Response 200 (application/vnd.api+json)
+
+  + Attributes (Post Response)
+
++ Response 404 (application/vnd.api+json)
+
+  + Attributes (Record Not Found Response)
+
+## POST [/posts/{id}]
```
Comment (npendery) [inline blueprint/api.apib]: Lets keep the titles consistent, so this should be `## Post [/posts/{id}]`

File: blueprint/api.apib [hunk 2]
```diff
@@ -141,6 +141,84 @@ Exchange credentials for `access_token`.
...
+
++ Response 404 (application/vnd.api+json)
+
+  + Attributes (Record Not Found Response)
+
++ Response 409 (application/vnd.api+json)
+
+  + Attributes (Creation Conflict Response)
+
+## POST COLLECTION [/posts]
```
Comment (npendery) [inline blueprint/api.apib]: For consistency, this should be `## Posts [/posts]`

Comment (joshsmith) [inline blueprint/api.apib]: And just `Posts`, no mention of collection, specifically, too.

File: blueprint/api.apib [hunk 3]
```diff
@@ -141,6 +141,84 @@ Exchange credentials for `access_token`.
...
+
+### List all posts [GET]
+
++ Headers
+
+  Authorization: Basic 5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8
+
++ Response 200 (application/vnd.api+json)
+
+  + Attributes (Posts Response)
```
Comment (npendery) [inline blueprint/api.apib]: Would you mind also adding the `Posts Response`?

Feel free to use the [categories response](https://github.com/code-corps/code-corps-api/pull/454/files#diff-788177756906fbee489264eddc07d051R302) as an example

File: blueprint/api.apib [hunk 4]
```diff
@@ -160,6 +160,84 @@ Exchange credentials for `access_token`.
...
+## Post [/posts]
+
+### Create a post [POST]
+
++ Request
+  + Attributes (OAuth Grant Request)
+
++ Headers
+
+  Authorization: Basic 5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8
```
Comment (npendery) [inline blueprint/api.apib]: Per the new style guide we are putting together for the API documentation (#463) would you mind changing these Authorization Headers to `Authorization: Bearer {access_token}`, thanks!

**Comment by joshsmith** [comment]
@madisonsites I've updated the ordering some here and fixed a bug. Make sure you preview at `localhost:8081`. There was a parse error. You can either run a linter locally or try to run something like Atom's preview to see if errors present.

Let me know if my changes make sense and you're okay with them.

**Comment by madisonsites** [comment]
All understood @JoshSmith ! Thank you!

**Comment by joshsmith** [comment]
@madisonsites need to suggest some stylistic changes based on some things I've discovered in writing more docs.

**Comment by joshsmith** [comment]
One thing we need to do is format each indented line by four spaces. This would go for things like `Attributes` for example.

File: blueprint/api.apib [hunk 5]
```diff
@@ -160,6 +160,84 @@ Exchange credentials for `access_token`.
...
+  + Attributes (Unprocessable Entity Response)
+
+## Post [/posts/{id}]
+
++ Parameters
+
+  + id: 1 (number, required)
+
+### Retrieve a post [GET]
+
```
Comment (joshsmith) [inline blueprint/api.apib:949]: You should change to `[GET /posts/{id}]`, move `Parameters` below that line, and remove the `## Post`.

File: blueprint/api.apib [hunk 6]
```diff
@@ -160,6 +160,84 @@ Exchange credentials for `access_token`.
...
+  + Attributes (Record Not Found Response)
+
+## POST [/posts/{id}]
+
++ Parameters
+
+  +  id: 1 (number, required)
+
+### Update a post [PUT]
```
Comment (joshsmith) [inline blueprint/api.apib]: You should change to `[PATCH /posts/{id}]`, move `Parameters` below that line, and remove the `## POST`.

File: blueprint/api.apib [hunk 7]
```diff
@@ -160,6 +160,84 @@ Exchange credentials for `access_token`.
...
+
+  + Attributes (Record Not Found Response)
+
++ Response 409 (application/vnd.api+json)
+
+  + Attributes (Creation Conflict Response)
+
+## Posts [/posts]
+
+### List all posts [GET]
```
Comment (joshsmith) [inline blueprint/api.apib]: You should change remove the `## Posts` line.

File: blueprint/api.apib [hunk 8]
```diff
@@ -581,6 +659,18 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
++ id: 1 (string)
++ status: `open` (string)
++ post_type: `task` (string)
++ title: `This is the title of a post` (string)
++ body: `This is the body of of a post filled with lots of awesome details` (string)
++ user_id: 1 (string)
+
+## Posts Response (object)
++ posts
+  + data(array[Post], required)
```
Comment (joshsmith) [inline blueprint/api.apib]: I think there are some changes to the response and resource objects that you can look to some of the recently closed PRs to.

**Comment by joshsmith** [comment]
@madisonsites you said you were going to tackle this again this evening, right?

**Comment by madisonsites** [comment]
@JoshSmith Yep, opening it up now

**Pull request merged**